### PR TITLE
code: add argument support

### PR DESCRIPTION
**Pull request by Victrid**
Add POSIX like arguments support for non-interactive use.
```
usage: main.py [-h] [-i | -I] [-r] [-o DOWNLOAD] [-c CONFIG_FILE] [--version] [-k]

Grab all files on Canvas LMS to local directory.

optional arguments:
  -h, --help            show this help message and exit
  -i, --interactive     Set the program to run in interactive mode (default action)
  -I, --non-interactive, --no-input
                        Set the program to run in non-interactive mode. This can be used to exit
                        immediately in case of profile corruption without getting stuck with the input.
  -r, --reconfigure, --configure
                        Reconfigure the tool.
  -o DOWNLOAD, --download-folder DOWNLOAD, --output DOWNLOAD
                        Set the download folder.
  -c CONFIG_FILE, --config-file CONFIG_FILE
                        Specify the configuration file. The configurations are loaded in this order:
                        arguments, file specified here, and the default config file config.toml.
  --version             show program's version number and exit
  -k, --keep-version, --no-update
                        Skip update checking. This will be helpful without a stable network connection
                        and prevent reconfiguration.
```
This will help automation with multiple profiles.

### CODE REVIEW

File: canvas_grab/__main__.py [hunk 1]
```diff
@@ -9,45 +9,92 @@
...
+            print(
+                "configuration file corrupted or not exist, and non interactive flag is set. Quit immediately.")
+            exit(-1)
         try:
             config.interact()
         except KeyboardInterrupt:
             print("User canceled the configuration process")
             return
-        Path('config.toml').write_text(
+        Path(args.config_file).write_text(
```
Comment (skyzh) [inline canvas_grab/__main__.py]: Could use `config_file` variable here.

File: canvas_grab/__main__.py [hunk 2]
```diff
@@ -9,45 +9,92 @@
...
+    interactive_group.add_argument("-I", "--non-interactive", "--no-input", dest="interactive", action="store_false", default=True,
+                                   help="Set the program to run in non-interactive mode. This can be used to exit immediately in case of profile corruption without getting stuck with the input.")
+
+    # Reconfiguration
+    parser.add_argument("-r", "--reconfigure", "--configure", dest="reconfigure", help="Reconfigure the tool.",
+                        action="store_true")
+
+    # Location Specification
+    parser.add_argument("-o", "--download-folder", "--output",
+                        dest="download", help="Set the download folder.")
```
Comment (skyzh) [inline canvas_grab/__main__.py]: What about using `override download folder` as help here, as `download_folder` is part of the configuration file?

**Review by skyzh — COMMENTED** [review]
Thanks for your contribution!

File: canvas_grab/__main__.py (hunk 2 above)
Comment (Victrid) [reply canvas_grab/__main__.py]: I don't think the word "override" should be used in this way. 
I changed it to `Specify alternative download folder.` instead.

File: canvas_grab/__main__.py [hunk 3]
```diff
@@ -79,7 +89,7 @@ def greeting():
         except KeyboardInterrupt:
             print("User canceled the configuration process")
             return
-        Path(args.config_file).write_text(
+        Path(config_file).write_text(
```
Comment (skyzh) [inline canvas_grab/__main__.py]: Seems that `config_file` itself is a Path object?

File: canvas_grab/__main__.py (hunk 2 above)
Comment (skyzh) [reply canvas_grab/__main__.py]: Alright

File: canvas_grab/__main__.py (hunk 3 above)
Comment (Victrid) [reply canvas_grab/__main__.py]: Forgot this 😅

**Review by skyzh — APPROVED** [review]

**Pull request merged**